Include/cpython/code.h

```diff
@@ -165,6 +172,12 @@ PyCode_NewEmpty(const char *filename, const char *funcname, int firstlineno);
    in this code object.  If you just need the line number of a frame,
    use PyFrame_GetLineNumber() instead. */
 PyAPI_FUNC(int) PyCode_Addr2Line(PyCodeObject *, int);
+/* Return the ending source code line number from a bytecode index. */
+PyAPI_FUNC(int) PyCode_Addr2EndLine(PyCodeObject*, int);
+/* Return the starting source code column offset from a bytecode index. */
+PyAPI_FUNC(int) PyCode_Addr2Offset(PyCodeObject*, int);
+/* Return the ending source code column offset from a bytecode index. */
+PyAPI_FUNC(int) PyCode_Addr2EndOffset(PyCodeObject*, int);
```

Comment:
One question, doesn't this expose these functions as part of the stable ABI? Is that intentional?

Reply:
It doesn't: `Include/cpython/*` is internal API.

Reply:
Ohh right, I forgot that for ``code.h`` in ``Python.h``, it's the one in ``Include/*``. Sorry and thanks for clarification Petr.

Reply:
@encukou  Is internal API but unless I am mistaken these are public functions. For instance, this file contains `PyCode_New` which is definitely public

Reply:
But in any case, yes, there is a new C-API function here (see the pep)